### ISSUE TITLE
Ever since vscode update my component hilighting is broken

### ISSUE THREAD

**Issue by maxwellsmart84** [vuejs/language-tools#4156]
<img width="720" alt="image" src="https://github.com/vuejs/language-tools/assets/14876902/cadb95f6-147c-4ab5-b4fb-562b185f1aec">

^^ Using Monokai, all Vue components are now green making it very difficult to read.

**Comment by YcZzy** [comment]
the same problem as you  just install the old version

**Comment by maxwellsmart84** [comment]
@YcZzy old version of VSCode or Volar?

**Comment by maxwellsmart84** [comment]
Downgrade to 2.6 did fix it for me.

<img width="610" alt="image" src="https://github.com/vuejs/language-tools/assets/14876902/8f895ae8-ce9f-4de4-a924-3e5161094a3d">

**Comment by davidmatter** [comment]
This is not a bug, it's just semantic highlighting working properly. You can remove it in the settings 
<img width="778" alt="image" src="https://github.com/vuejs/language-tools/assets/2319523/ba1708f8-1a23-41df-be1d-520c08e3abf3">

**Comment by Laurelin** [comment]
This is not semantic highlighting working properly; it doesn't recognize components at all. have to revert to 1.8.27 to fix it. Big problem, because it doesn't know what props my components need, makes breaking things inevitable.

**Comment by CyberCowboy404** [comment]
I've noticed that my highlight broke after I run code formatting.
After formatting my script section becomes like this
```
<script
  setup
  lang='ts'
>
```

This is annoying behavior which I also got after update.
i suppose it may cause such problem

**Comment by killjoy1221** [comment]
The highlighting isn't completely broken for me using Dark Modern theme, but it does blink in and out sometimes as I scroll. Specifically, I've seen that if I start to scroll past about 50 lines of `<template>` code, the highlighting breaks.

**Comment by qiaozhu** [comment]
切换折叠、滚动条滚动等情况下，组件高亮会闪烁

**Comment by XioDone** [comment]
@johnsoncodehk
Is it possible to refer to #2252 to fix it?
<img width="599" alt="image" src="https://github.com/vuejs/language-tools/assets/82227884/f111c694-323e-474c-aec6-f79c10aabd72">

**Comment by johnsoncodehk** [comment]
@XioDone No, with TS plugin it can't use "component" semantic token type.

**Comment by DocMcCoy** [comment]
I've been annoyed with this for a minute but while I was messing around I noticed highlighting changing as I changed the lang attribute.

<img width="372" alt="Screenshot 2024-04-16 at 4 08 50 PM" src="https://github.com/vuejs/language-tools/assets/43177751/9624985b-fe11-4220-87e0-86799e42691a">

<img width="362" alt="Screenshot 2024-04-16 at 4 10 34 PM" src="https://github.com/vuejs/language-tools/assets/43177751/a14f3412-7eb9-4b35-9c7c-a26224886dac">

"js" doesn't seem to be the lang that you would want there however. Also interesting is that in both scenarios the built-in TransitionGroup component is highlighted as expected.

**Comment by amfischer** [comment]
> No, with TS plugin it can't use "component" semantic token type.

@johnsoncodehk  Does this mean custom components should have the same highlight color as normal html tags?

> I've been annoyed with this for a minute but while I was messing around I noticed highlighting changing as I changed the lang attribute.

@DocMcCoy interesting, adding `lang="js` to my template tag does correct the highlighting but causes eslint to throw errors in my `<script>` section.

**Comment by so1ve** [comment]
@amfischer Internally components are converted to functions. So they should have the same color as functions

**Comment by amateuz** [comment]
Faced this issue. Rather annoying for me.
@amfischer Same here. Adding lang="js" to my template fixes the issue but eslint starts to throw errors.

**Comment by jesspinklet** [comment]
Same here, after upgrading vscode currently is => 1.96.4 I lost the component highlighting. The types and auto complete works tho but still it's very annoying for me
